src/libs/functions/urlHelpers.js

```diff
@@ -40,7 +40,41 @@ export const checkIsRedirectOutside = ({ currentUrl, destinationUrl }) => {
 
   return false
 }
+/**
+ *
+ * @param {object} params
+ * @param {string} params.cozyUrl The URL of the cozy, from client.stackclient.uri
+ * @param {string} params.destinationUrl The URL we want to redirect the user
+ * @param {'flat' | 'nested'} params.subdomainType - the Cozy's subdomain type
+ * @returns {boolean} True is the destination is not on the cozy
+ */
+export const checkIsCozyUrlIsTheDestination = ({
+  cozyUrl,
+  destinationUrl,
+  subdomainType = 'flat'
+}) => {
+  try {
+    const currentUrlData = deconstructCozyWebLinkWithSlug(
+      cozyUrl,
+      subdomainType
+    )
+    const destinationUrlData = deconstructCozyWebLinkWithSlug(
+      destinationUrl,
+      subdomainType
+    )
+    if (
+      currentUrlData.cozyBaseDomain !== destinationUrlData.cozyBaseDomain ||
+      currentUrlData.cozyName !== destinationUrlData.cozyName
+    ) {
+      return true
+    }
 
+    return false
+  } catch (err) {
+    log.error('Error while calling checkIsCozyUrlIsTheDestination', err)
+    return false
```

Comment:
I would return `true` instead.

If we don't succeed to compare URL I prefer to have a false positive and open IAB than the reverse. This would prevent the user to be stuck in a random url.
Also the more common error source would be an exception while parsing the destination URL, which should happen only if it is from another server with different subdomain type (or just a badly formated URL)

Reply:
Oh you right, nice catch